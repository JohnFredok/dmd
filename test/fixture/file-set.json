[
  {
    "id": "module:file-set",
    "longname": "module:file-set",
    "name": "file-set",
    "kind": "module",
    "description": "Exports a contructor taking a list of file patterns as input, returning a `file-set` instance containing the expanded patterns split into separate lists of `files`, `dirs` and `notExisting`.",
    "examples": [
      "```js\nvar fileSet = require(\"file-set\");\n```"
    ],
    "typicalname": "FileSet",
    "order": 0
  },
  {
    "id": "module:file-set--FileSet",
    "longname": "module:file-set",
    "name": "FileSet",
    "kind": "class",
    "isExported": true,
    "description": "Expands file patterns, returning the matched and unmatched files and directories",
    "memberof": "module:file-set",
    "alias": "module:file-set",
    "codeName": "FileSet",
    "order": 1
  },
  {
    "id": "module:file-set--FileSet()",
    "longname": "module:file-set",
    "name": "FileSet",
    "kind": "constructor",
    "memberof": "module:file-set--FileSet",
    "params": [
      {
        "type": {
          "names": [
            "string",
            "Array.<string>"
          ]
        },
        "description": "A pattern, or array of patterns to expand",
        "name": "patternList"
      }
    ],
    "order": 2
  },
  {
    "id": "module:file-set--FileSet#list",
    "longname": "module:file-set#list",
    "name": "list",
    "scope": "instance",
    "kind": "member",
    "description": "The full list of unique paths found, and not found.",
    "memberof": "module:file-set--FileSet",
    "type": {
      "names": [
        "Array.<string>"
      ]
    },
    "codeName": "this.list",
    "order": 3
  },
  {
    "id": "module:file-set--FileSet#files",
    "longname": "module:file-set#files",
    "name": "files",
    "scope": "instance",
    "kind": "member",
    "description": "The existing files found",
    "memberof": "module:file-set--FileSet",
    "type": {
      "names": [
        "Array.<string>"
      ]
    },
    "codeName": "this.files",
    "order": 4
  },
  {
    "id": "module:file-set--FileSet#dirs",
    "longname": "module:file-set#dirs",
    "name": "dirs",
    "scope": "instance",
    "kind": "member",
    "description": "The existing directories found",
    "memberof": "module:file-set--FileSet",
    "type": {
      "names": [
        "Array.<string>"
      ]
    },
    "codeName": "this.dirs",
    "order": 5
  },
  {
    "id": "module:file-set--FileSet#notExisting",
    "longname": "module:file-set#notExisting",
    "name": "notExisting",
    "scope": "instance",
    "kind": "member",
    "description": "Paths which were not found",
    "memberof": "module:file-set--FileSet",
    "type": {
      "names": [
        "Array.<string>"
      ]
    },
    "codeName": "this.notExisting",
    "order": 6
  },
  {
    "id": "module:file-set--FileSet#add",
    "longname": "module:file-set#add",
    "name": "add",
    "scope": "instance",
    "kind": "function",
    "description": "add file patterns to the set",
    "memberof": "module:file-set--FileSet",
    "params": [
      {
        "type": {
          "names": [
            "string",
            "Array.<string>"
          ]
        },
        "description": "A pattern, or array of patterns to expand",
        "name": "files"
      }
    ],
    "codeName": "FileSet.prototype.add",
    "order": 7
  },
  {
    "id": "module:file-set--FileSet.eFileType",
    "longname": "module:file-set.eFileType",
    "name": "eFileType",
    "scope": "static",
    "kind": "member",
    "description": "Enum for the `type` value of each record in `fileSet.list`",
    "memberof": "module:file-set--FileSet",
    "type": {
      "names": [
        "number"
      ]
    },
    "readonly": true,
    "isEnum": true,
    "properties": [
      {
        "description": "when a file doesn't exist",
        "name": "NOEXIST",
        "longname": "module:file-set.eFileType.NOEXIST",
        "kind": "member",
        "memberof": "module:file-set.eFileType",
        "scope": "static",
        "type": {
          "names": [
            "number"
          ]
        },
        "defaultvalue": "0",
        "codeName": "NOEXIST"
      },
      {
        "description": "It's a file",
        "name": "FILE",
        "longname": "module:file-set.eFileType.FILE",
        "kind": "member",
        "memberof": "module:file-set.eFileType",
        "scope": "static",
        "type": {
          "names": [
            "number"
          ]
        },
        "defaultvalue": "1",
        "codeName": "FILE"
      },
      {
        "name": "DIR",
        "longname": "module:file-set.eFileType.DIR",
        "kind": "member",
        "memberof": "module:file-set.eFileType",
        "scope": "static",
        "type": {
          "names": [
            "number"
          ]
        },
        "defaultvalue": "2",
        "codeName": "DIR"
      }
    ],
    "codeName": "FileSet.eFileType",
    "order": 8
  }
]