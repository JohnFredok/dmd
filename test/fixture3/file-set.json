[
  {
    "description": "Exports a contructor taking a list of file patterns as input, returning a `file-set` instance containing the expanded patterns split into separate lists of `files`, `dirs` and `notExisting`.",
    "kind": "module",
    "name": "file-set",
    "examples": [
      "```js\nvar fileSet = require(\"file-set\");\n```"
    ],
    "id": "module:file-set"
  },
  {
    "kind": "class",
    "classdesc": "Expands file patterns, returning the matched and unmatched files and directories",
    "name": "FileSet",
    "id": "module:file-set^FileSet",
    "memberof": "module:file-set",
    "codeName": "FileSet"
  },
  {
    "kind": "constructor",
    "params": [
      {
        "type": {
          "names": [
            "string",
            "Array.<string>"
          ]
        },
        "description": "A pattern, or array of patterns to expand",
        "name": "patternList"
      }
    ],
    "name": "FileSet",
    "id": "module:file-set^newFileSet",
    "memberof": "module:file-set^FileSet",
    "codeName": "FileSet"
  },
  {
    "description": "The full list of unique paths found, and not found.",
    "type": {
      "names": [
        "Array.<string>"
      ]
    },
    "name": "list",
    "id": "module:file-set#list",
    "kind": "member",
    "memberof": "module:file-set^FileSet",
    "scope": "instance",
    "codeName": "this.list"
  },
  {
    "description": "The existing files found",
    "type": {
      "names": [
        "Array.<string>"
      ]
    },
    "name": "files",
    "id": "module:file-set#files",
    "kind": "member",
    "memberof": "module:file-set^FileSet",
    "scope": "instance",
    "codeName": "this.files"
  },
  {
    "description": "The existing directories found",
    "type": {
      "names": [
        "Array.<string>"
      ]
    },
    "name": "dirs",
    "id": "module:file-set#dirs",
    "kind": "member",
    "memberof": "module:file-set^FileSet",
    "scope": "instance",
    "codeName": "this.dirs"
  },
  {
    "description": "Paths which were not found",
    "type": {
      "names": [
        "Array.<string>"
      ]
    },
    "name": "notExisting",
    "id": "module:file-set#notExisting",
    "kind": "member",
    "memberof": "module:file-set^FileSet",
    "scope": "instance",
    "codeName": "this.notExisting"
  },
  {
    "description": "add file patterns to the set",
    "params": [
      {
        "type": {
          "names": [
            "string",
            "Array.<string>"
          ]
        },
        "description": "A pattern, or array of patterns to expand",
        "name": "files"
      }
    ],
    "name": "add",
    "id": "module:file-set#add",
    "kind": "function",
    "memberof": "module:file-set^FileSet",
    "scope": "instance",
    "codeName": "FileSet.prototype.add"
  },
  {
    "description": "Enum for the `type` value of each record in `fileSet.list`",
    "readonly": true,
    "kind": "member",
    "isEnum": true,
    "type": {
      "names": [
        "number"
      ]
    },
    "name": "eFileType",
    "id": "module:file-set.eFileType",
    "memberof": "module:file-set^FileSet",
    "scope": "static",
    "properties": [
      {
        "scope": "static",
        "type": {
          "names": [
            "number"
          ]
        },
        "memberof": "module:file-set.eFileType",
        "id": "module:file-set.eFileType.NOEXIST",
        "name": "NOEXIST",
        "kind": "member",
        "defaultvalue": "0",
        "codeName": "NOEXIST"
      },
      {
        "scope": "static",
        "type": {
          "names": [
            "number"
          ]
        },
        "memberof": "module:file-set.eFileType",
        "id": "module:file-set.eFileType.FILE",
        "name": "FILE",
        "kind": "member",
        "defaultvalue": "1",
        "codeName": "FILE"
      },
      {
        "scope": "static",
        "type": {
          "names": [
            "number"
          ]
        },
        "memberof": "module:file-set.eFileType",
        "id": "module:file-set.eFileType.DIR",
        "name": "DIR",
        "kind": "member",
        "defaultvalue": "2",
        "codeName": "DIR"
      }
    ],
    "codeName": "FileSet.eFileType"
  },
  {
    "name": "NOEXIST",
    "id": "module:file-set.eFileType.NOEXIST",
    "kind": "member",
    "memberof": "module:file-set.eFileType",
    "scope": "static",
    "type": {
      "names": [
        "number"
      ]
    },
    "defaultvalue": "0",
    "codeName": "NOEXIST"
  },
  {
    "name": "FILE",
    "id": "module:file-set.eFileType.FILE",
    "kind": "member",
    "memberof": "module:file-set.eFileType",
    "scope": "static",
    "type": {
      "names": [
        "number"
      ]
    },
    "defaultvalue": "1",
    "codeName": "FILE"
  },
  {
    "name": "DIR",
    "id": "module:file-set.eFileType.DIR",
    "kind": "member",
    "memberof": "module:file-set.eFileType",
    "scope": "static",
    "type": {
      "names": [
        "number"
      ]
    },
    "defaultvalue": "2",
    "codeName": "DIR"
  }
]