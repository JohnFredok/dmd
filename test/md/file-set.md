<a name="module_file-set"></a>
##file-set
Exports a contructor taking a list of file patterns as input, returning a `file-set` instance containing the expanded patterns split into separate lists of `files`, `dirs` and `notExisting`.

**Example**  
```js
var fileSet = require("file-set");
```

* [file-set](#module_file-set)
  * [class: FileSet](#exp_module_file-set--FileSet) ⏏
    * [new FileSet(patternList)](#new_module_file-set--FileSet_new)
    * _instance_
      * [.list](#module_file-set--FileSet#list) → <code>Array.&lt;string&gt;</code>
      * [.files](#module_file-set--FileSet#files) → <code>Array.&lt;string&gt;</code>
      * [.dirs](#module_file-set--FileSet#dirs) → <code>Array.&lt;string&gt;</code>
      * [.notExisting](#module_file-set--FileSet#notExisting) → <code>Array.&lt;string&gt;</code>
      * [.add(files)](#module_file-set--FileSet#add)
    * _static_
      * [enum: .eFileType](#module_file-set--FileSet.eFileType) → <code>number</code>

<a name="exp_module_file-set--FileSet"></a>
###class: FileSet ⏏
Expands file patterns, returning the matched and unmatched files and directories

<a name="new_module_file-set--FileSet_new"></a>
####new FileSet(patternList)
| Param | Type | Description |
| ----- | ---- | ----------- |
| patternList | <code>string</code> \| <code>Array.&lt;string&gt;</code> | A pattern, or array of patterns to expand |

<a name="module_file-set--FileSet#list"></a>
####fileSet.list → <code>Array.&lt;string&gt;</code>
The full list of unique paths found, and not found.

<a name="module_file-set--FileSet#files"></a>
####fileSet.files → <code>Array.&lt;string&gt;</code>
The existing files found

<a name="module_file-set--FileSet#dirs"></a>
####fileSet.dirs → <code>Array.&lt;string&gt;</code>
The existing directories found

<a name="module_file-set--FileSet#notExisting"></a>
####fileSet.notExisting → <code>Array.&lt;string&gt;</code>
Paths which were not found

<a name="module_file-set--FileSet#add"></a>
####fileSet.add(files)
add file patterns to the set

| Param | Type | Description |
| ----- | ---- | ----------- |
| files | <code>string</code> \| <code>Array.&lt;string&gt;</code> | A pattern, or array of patterns to expand |

<a name="module_file-set--FileSet.eFileType"></a>
####enum: FileSet.eFileType → <code>number</code>
Enum for the `type` value of each record in `fileSet.list`

**Properties**

| Name | Type | Default |
| --- | --- | --- |
| NOEXIST | <code>number</code> | `0` | 
| FILE | <code>number</code> | `1` | 
| DIR | <code>number</code> | `2` | 

**Read only**: true  
