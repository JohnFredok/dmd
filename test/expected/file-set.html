<!DOCTYPE html>
<html>
  <head>
    <title>Docs</title>
    <meta charset="utf8">
    <link rel="stylesheet" href="/html/docs.css">
    <link rel="stylesheet" href="/html/coy.css">
  </head>
  <body class="overflow-hidden">
    <div class="flexbox-col box-fill-viewport">
      <div class="row1 flex-fixed">
        <h1>API Reference</h1>
      </div>
      <div class="row2 flexbox-row">
        <div class="col1 horiz-space flex-fixed overflow-scroll">
          <h3>Modules</h3><ul><li>
  <a href="#module_file-set">module: file-set</a>
  
    <ul>
    <li>
  <a href="#exp_module_file-set^FileSet">class: FileSet ⏏</a>
  
    <ul>
    <li>
  <a href="#new_module_file-set^FileSet◊">new FileSet(patternList)</a>
  
</li>

  
    
    <li>
  <a href="#module_file-set^FileSet#list">fileSet.list</a>
  
</li>

  
    
    <li>
  <a href="#module_file-set^FileSet#files">fileSet.files</a>
  
</li>

  
    
    <li>
  <a href="#module_file-set^FileSet#dirs">fileSet.dirs</a>
  
</li>

  
    
    <li>
  <a href="#module_file-set^FileSet#notExisting">fileSet.notExisting</a>
  
</li>

  
    
    <li>
  <a href="#module_file-set^FileSet#add">fileSet.add(files)</a>
  
</li>

  
    
    <li>
  <a href="#module_file-set^FileSet.eFileType">FileSet.eFileType</a>
  
</li>
</ul>
  
</li>
</ul>
  
</li>
</ul>
          

          

          
        </div>
        <div class="col2 overflow-scroll flex-full">
        
          <h2><a name="module_file-set">module: file-set</a></h2>
<p>Exports a contructor taking a list of file patterns as input, returning a <code>file-set</code> instance containing the expanded patterns split into separate lists of <code>files</code>, <code>dirs</code> and <code>notExisting</code>.</p>





<strong>Example</strong>
<pre><code class="language-javascript">var fileSet = require(&quot;file-set&quot;);
</code></pre>



<dl>
  <dt>kind</dt><dd>module</dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <dt>typicalname</dt><dd>FileSet</dd>
  
  
</dl>

<h3><a name="exp_module_file-set^FileSet">class: FileSet ⏏</a></h3>
<p>Expands file patterns, returning the matched and unmatched files and directories</p>






<dl>
  <dt>kind</dt><dd>class</dd>
  
  <dt>memberof</dt><dd><code> class of <a href="#module_file-set">file-set</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="new_module_file-set^FileSet◊">new FileSet(patternList)</a></h4>


<strong>Params</strong>
<ul>
<li>patternList <code>string,Array.&lt;string&gt;</code> A pattern, or array of patterns to expand</li>
</ul>




<dl>
  <dt>kind</dt><dd>constructor</dd>
  
  <dt>memberof</dt><dd><code> constructor of <a href="#exp_module_file-set^FileSet">FileSet</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_file-set^FileSet#list">fileSet.list</a></h4>
<p>The full list of unique paths found, and not found.</p>






<dl>
  <dt>kind</dt><dd>member</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance member of <a href="#exp_module_file-set^FileSet">FileSet</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_file-set^FileSet#files">fileSet.files</a></h4>
<p>The existing files found</p>






<dl>
  <dt>kind</dt><dd>member</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance member of <a href="#exp_module_file-set^FileSet">FileSet</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_file-set^FileSet#dirs">fileSet.dirs</a></h4>
<p>The existing directories found</p>






<dl>
  <dt>kind</dt><dd>member</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance member of <a href="#exp_module_file-set^FileSet">FileSet</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_file-set^FileSet#notExisting">fileSet.notExisting</a></h4>
<p>Paths which were not found</p>






<dl>
  <dt>kind</dt><dd>member</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance member of <a href="#exp_module_file-set^FileSet">FileSet</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_file-set^FileSet#add">fileSet.add(files)</a></h4>
<p>add file patterns to the set</p>


<strong>Params</strong>
<ul>
<li>files <code>string,Array.&lt;string&gt;</code> A pattern, or array of patterns to expand</li>
</ul>




<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance function of <a href="#exp_module_file-set^FileSet">FileSet</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_file-set^FileSet.eFileType">FileSet.eFileType</a></h4>
<p>Enum for the <code>type</code> value of each record in <code>fileSet.list</code></p>






<dl>
  <dt>kind</dt><dd>member</dd>
  <dt>scope</dt><dd>static</dd>
  <dt>memberof</dt><dd><code>static member of <a href="#exp_module_file-set^FileSet">FileSet</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>


        </div>
      </div>
    </div>
    <script src="/html/prism.js"></script>
  </body>
</html>
